### ISSUE TITLE
IMPORTANT: SEE HERE FOR CONTROLLER PROBLEMS ON QUEST

### ISSUE THREAD

**Issue by krzys-h** [polygraphene/ALVR#491]
Since @polygraphene is nowhere to be seen, I'm leaving this here for people who are coming here and asking this question over and over since I have no good resource to point them to. #479 #477 #476 #484

A bit over a week ago, a firmware update for Oculus Quest was released that apparently introduced some kind of compatibility for Oculus Go / Gear VR which activates if the app does not declare the `android.hardware.vr.headtracking` feature. In the past it was only really required when submitting the app to the store to pass certification and missing it out didn't break anything, hence why most sideloaded apps failed to do it properly.

The compatibility mode allows only one controller to be active at a time and makes it appear to the program as the [Oculus Go controller](https://developer.oculus.com/documentation/unity/latest/concepts/unity-ovrinput/#oculus-go-controller-input-handling) rather than the normal Quest one (which also limits the available input buttons)

I submitted a pull request to fix it in https://github.com/polygraphene/ALVRClient/pull/32, but until @polygraphene comes back to merge it and make an official release, you will have to get a patched version from somewhere. I've seen a few of these around, but since I don't generally trust random binaries from the internet so here is a build I made myself from the master branch with this fix included: https://github.com/krzys-h/ALVRClient/releases/tag/v2.4.0-alpha5-quest-fix

---

**tl;dr**, if you are not a developer and don't care about the technical things, here is what to do to get it to work again. I'll assume you already know how to use `adb` since you somehow managed to install the official, broken version (if not, see https://github.com/polygraphene/ALVR/wiki/Installation)
1. Download a patched version from https://github.com/krzys-h/ALVRClient/releases/tag/v2.4.0-alpha5-quest-fix
2. `adb uninstall com.polygraphene.alvr` (this has to be done explicitly since the patched version uses different signing keys because obviously nobody except @polygraphene has the official ones)
3. `adb install path-to-patched-version.apk`

PS. Apparently somebody is working on a version which contains this fix as well as some other improvements to controller tracking (see #482) although I haven't tested it so YMMV

**Comment by Irtharan** [comment]
@krzys-h Thanks for the reference, but this actually has nothing to do with what I meant =P I knew about that one error, mine is a completely different one which was there even before the update for the "Go" compatibility.

Anyway, as I said, thanks for the reference and the post itself, which I'm sure it'll help lots of users.

**Comment by krzys-h** [comment]
@Irtharan I know yours is different, the reason it appeared to you is because I put a link to the fork with various improvements at the end of the post

**Comment by Irtharan** [comment]
@krzys-h Ohh okay, sorry. Totally new into github, so dunno about how it totally works still ^^'

**Comment by yackaro** [comment]
Wish I'd seen this post two days ago, I already made my own patched apk for a friend X_X

**Comment by cowsiwin** [comment]
Thanks for taking the time to do this.  Works as expected all the controls in Google Earth work as expected.

**Comment by Jacktherookie** [comment]
Wow. Thanks for the revised version. However, While I was using my Oculus Touch Controller (Not Oculus Rift one), Oculus Rift controller appears rather than Oculus Touch controller (Bundled with Quest) in SteamVR Home (And also some kind of mismatch happens at controller tracking). But when I use SteamVR overlay and any other apps (like SteamVR tutorial) controllers seems fine. I dont know why this happens.

**Comment by tooandrew** [comment]
this version linked is not seen by the alvr server, while the official version was found, but controllers didnt work

**Comment by LexxLoon** [comment]
Hi everyone,
is there a newer version? This doesn't seem to work. I am connected to the server, I am in the Steam Home but do not see any controllers or even hands.
I installed
"app-Github-Normal-Ovr-debug.apk" on the glasses and
"v2.4.0-alpha5" server app

greetings
the Lexx

**Comment by LilSizzurp** [comment]
images very far apart, help?

**Comment by secretsquirrel420** [comment]
> 
> 
> Hi everyone,
> is there a newer version? This doesn't seem to work. I am connected to the server, I am in the Steam Home but do not see any controllers or even hands.
> I installed
> "app-Github-Normal-Ovr-debug.apk" on the glasses and
> "v2.4.0-alpha5" server app
> 
> greetings
> the Lexx

Are you certain that you do not have the long press enabled?

See: https://www.reddit.com/r/oculus/comments/awv1iv/oculus_touch_controller_oculushome_button_not/

**Comment by super-panda117** [comment]
im still having trouble with the controllers not working once i'm in game. works in the steam vr home but not in games. help? trying to play some games, maybe even ALYX  in quarantine

**Comment by super-panda117** [comment]
> im still having trouble with the controllers not working once i'm in game. works in the steam vr home but not in games. help? trying to play some games, maybe even ALYX in quarantine

i also installed the fix as suggested to no avail. programming
 noob here.
